sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningUtils.scala

```diff
@@ -351,8 +354,10 @@ private[sql] object PartitioningUtils {
       }
     }
 
-    if (partitionColumns.size == schema.fields.size) {
-      throw new AnalysisException(s"Cannot use all columns for partition columns")
+    if (schema.fields.isEmpty) {
```

Comment:
Thank you for review, @tdas.
Yes. Indeed. This is beyond of the scope of `PartitioningUtils`.
Actually, this logic is used 3 classes, `PreWriteCheck.`, `DataSource`, `FileStreamSinkWriter`
I'll try to move this.